lib/chef/provider/windows_task.rb

```diff
@@ -0,0 +1,396 @@
+      end
+
+      def update_task_xml(options = [])
+        # random_delay xml element is different for different frequencies
+        random_delay_xml_element = {
+          :minute => "Triggers/TimeTrigger/RandomDelay",
+          :hourly => "Triggers/TimeTrigger/RandomDelay",
+          :once => "Triggers/TimeTrigger/RandomDelay",
+          :daily => "Triggers/CalendarTrigger/RandomDelay",
+          :weekly => "Triggers/CalendarTrigger/RandomDelay",
+          :monthly => "Triggers/CalendarTrigger/RandomDelay",
+        }
+
+        xml_element_mapping = {
+          "cwd" => "Actions/Exec/WorkingDirectory",
+          "random_delay" => random_delay_xml_element[@new_resource.frequency],
+          "execution_time_limit" => "Settings/ExecutionTimeLimit"
+        }
+
+        Chef::Log.debug 'looking for existing tasks'
+
+        # we use shell_out here instead of shell_out! because a failure implies that the task does not exist
+        xml_cmd = shell_out("schtasks /Query /TN \"#{@new_resource.task_name}\" /XML")
+
+        return if xml_cmd.exitstatus != 0
+
+        doc = REXML::Document.new(xml_cmd.stdout)
+
+        options.each do |option|
+          Chef::Log.debug 'Removing former #{option} if any'
+          doc.root.elements.delete(xml_element_mapping[option])
+          option_value = @new_resource.send("#{option}")
+
+          if option_value
+            Chef::Log.debug 'Setting #option as #option_value'
+            split_xml_path = xml_element_mapping[option].split("/")  # eg. if xml_element_mapping[option] = "Actions/Exec/WorkingDirectory"
+            element_name = split_xml_path.last   # element_name = "WorkingDirectory"
+            cwd_element = REXML::Element.new(element_name)
+            cwd_element.add_text(option_value)
+            element_root = (split_xml_path - [element_name]).join("/") # element_root = 'Actions/Exec'
+            exec_element = doc.root.elements[element_root]
+            exec_element.add_element(cwd_element)
+          end
+        end
+
+        temp_task_file = ::File.join(ENV['TEMP'], 'windows_task.xml')
+        begin
+          ::File.open(temp_task_file, 'w:UTF-16LE') do |f|
+            doc.write(f)
+          end
+
+          options = {}
+          options['RU'] = @new_resource.user if @new_resource.user
+          options['RP'] = @new_resource.password if @new_resource.password
+          options['IT'] = '' if @new_resource.interactive_enabled
+          options['XML'] = temp_task_file
+
+          run_schtasks('DELETE', 'F' => '')
+          run_schtasks('CREATE', options)
+        ensure
+          ::File.delete(temp_task_file)
+        end
+      end
+
+      def load_task_hash(task_name)
+        Chef::Log.debug 'Looking for existing tasks'
+
+        # we use shell_out here instead of shell_out! because a failure implies that the task does not exist
+        output = shell_out("schtasks /Query /FO LIST /V /TN \"#{task_name}\"").stdout

```

Comment:
If Windows locale is different than English, for example, Russian, the output will be in Russian language so the hash will be completely useless. Therefore, resource will be non-idempotent by design.
As a workaround we can use chcp 437 prior to run schtasks.exe
```cmd
output = shell_out("chcp 437 && schtasks /Query /FO LIST /V /TN \"#{task_name}\"").stdout
```